types/semver/index.d.ts

```diff
@@ -113,6 +113,9 @@ export type ReleaseType = 'major' | 'premajor' | 'minor' | 'preminor' | 'patch'
 
 export interface Options {
     loose?: boolean | undefined;
+}
+
+export interface RangeOptions extends Options {
     includePrerelease?: boolean | undefined;
 }
```

Comment:
This is a breaking change if people use `Semver.Options` right now, but I couldn't find a better name. There are no "general" options in the `semver` package. Open for suggestions!

Reply:
aren't all options supposed to support this prop? Say:
https://github.com/npm/node-semver/blob/802e16174fe2a704dba16e97891ce36dc4f2ad76/classes/semver.js#L33
('false' by default, can be boolean or an object with `loose` or `includePrerelease`)

Reply:
Hey @peterblazejewicz! Thanks for your question.

As for the code you linked, you can see the [comment above](https://github.com/npm/node-semver/blob/802e16174fe2a704dba16e97891ce36dc4f2ad76/classes/semver.js#L31-L33), saying:

> this isn't actually relevant for versions

While the constructor takes the additional argument (as it does in this PR), if you import the functions directly, they are not needed, omitted and have no effect. By being more strict with the types, we can avoid pitfalls.

For example, let's take the `compare` function:

https://github.com/npm/node-semver/blob/802e16174fe2a704dba16e97891ce36dc4f2ad76/functions/compare.js

It uses the constructor, but is only interested in `loose`.

When checking the tests, we can also only see that it's being tested with a `boolean` and an object: `{ loose: loose }`:

https://github.com/npm/node-semver/blob/802e16174fe2a704dba16e97891ce36dc4f2ad76/test/functions/compare.js

The fixtures for the test also only ever provide a `loose` argument:

https://github.com/npm/node-semver/blob/802e16174fe2a704dba16e97891ce36dc4f2ad76/test/fixtures/comparisons.js

And as their README says:

> setting the `includePrerelease` flag on the options object to any [functions](https://github.com/npm/node-semver/tree/v7.3.7#functions) that **do range matching**.

I'll take `satisfies` as an example, linking only the fixtures in this case. It takes no options, a `boolean`, and an object, namely what I called `RangeOptions`:

https://github.com/npm/node-semver/blob/v7.3.7/test/fixtures/range-exclude.js
https://github.com/npm/node-semver/blob/v7.3.7/test/fixtures/range-include.js

Does that answer your question?

Reply:
This makes sense to me.

Reply:
Removing `includePrerelease` from the generic Options seems to go against the documentation - https://github.com/npm/node-semver/blob/main/README.md?plain=1#L364-L378

>All methods and classes take a final `options` object argument.  All
options in this object are `false` by default.  The options supported
are:
> - `loose`  ...
> - `includePrerelease`  ...

Would you say that the documentation is misleading about the actual effect of the `includePrerelease` flag?